File: openclaw-userlook/frontend/src/api/wecom.ts
import { apiClient } from './client'
import type { LoginResponse } from './auth'

export interface WeComLoginUrlResponse {
  login_url: string
  mock: boolean
  message?: string
}

export interface WeComBinding {
  provider: string
  external_user_id?: string
  external_open_id?: string
  external_union_id?: string
  display_name?: string
  created_at: string
}

export interface WeComMeResponse {
  bound: boolean
  login_source: 'wecom'
  binding: WeComBinding | null
}

export async function fetchWeComLoginUrl(state = ''): Promise<WeComLoginUrlResponse> {
  const response = await apiClient.get<WeComLoginUrlResponse>('/api/wecom/login-url', {
    params: state ? { state } : undefined,
  })
  return response.data
}

export async function finishWeComLogin(code: string, state: string): Promise<LoginResponse> {
  const response = await apiClient.get<LoginResponse>('/api/wecom/callback', {
    params: { code, state },
  })
  return response.data
}

export async function fetchWeComMe(): Promise<WeComMeResponse> {
  const response = await apiClient.get<WeComMeResponse>('/api/wecom/me')
  return response.data
}

File: openclaw-userlook/backend/app/schemas/wecom.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.user import UserRead


class WeComLoginUrlResponse(BaseModel):
    login_url: str
    mock: bool
    message: str | None = None


class WeComBindingRead(BaseModel):
    provider: str
    external_user_id: str | None = None
    external_open_id: str | None = None
    external_union_id: str | None = None
    display_name: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WeComCallbackResponse(BaseModel):
    access_token: str
    token_type: str = "bearer"
    user: UserRead
    binding: WeComBindingRead
    login_source: str = "wecom"

    model_config = ConfigDict(from_attributes=True)


class WeComMeResponse(BaseModel):
    bound: bool
    login_source: str = "wecom"
    binding: WeComBindingRead | None = None
